iree/base/internal/file_io_win32.cc

```diff
@@ -87,6 +90,44 @@ Status MoveFile(const std::string& source_path,
   return OkStatus();
 }
 
+std::string GetTempPath() {
+  IREE_TRACE_SCOPE0("file_io::GetTempPath");
+
+  // TEST_TMPDIR will point to a writeable temp path when running bazel tests.
+  char* test_tmpdir = getenv("TEST_TMPDIR");
+  if (test_tmpdir) {
+    return test_tmpdir;
+  }
+
+  DWORD chars_required = ::GetTempPathA(0, nullptr);
```

Comment:
not that performance is particularly an issue here, but a good habit to get into with this style of query is to presize to a reasonable value and try that first, only doing another query when needed.

For example, this removes the need to query twice in the common case and removes the duplicate error handling:

```
std::string temp_path(64, '\0');
for (bool retry_query = false; retry_query && !temp_path.empty();) {
  DWORD required_length = ::GetTempPathA(temp_path.size(), temp_path.data());
  retry_query = required_length > temp_path.size();
  temp_path.resize(required_length);
}
return temp_path;
```

(realistically GetTempPath will never fail, so dchecking and such is overkill)

Reply:
👍 
(`temp_path.data()` -> `&temp_path[0]` since `GetTempPathA` modifies the string)